modules/build/src/main/scala/scala/build/preprocessing/ExtractedDirectives.scala

```diff
@@ -18,16 +18,29 @@ import scala.jdk.CollectionConverters.*
 
 case class ExtractedDirectives(
   offset: Int,
-  directives: Seq[StrictDirective]
+  directives: Seq[StrictDirective],
+  location: Option[DirectivesLocation]
 ) {
   @targetName("append")
   def ++(other: ExtractedDirectives): ExtractedDirectives =
-    ExtractedDirectives(offset, directives ++ other.directives)
+    ExtractedDirectives(offset, directives ++ other.directives, location)
 }
 
+case class DirectivesLocation(
+  codeDirectives: Location,
+  specialCommentDirectives: Location,
+  plainCommentDirectives: Location
+)
+
+case class Location(
```

Comment:
Why not use `Position.File` instead of `Location`?

Reply:
I didn't think about using `Position.File`. It seems logical. I will use it then.